tutor/src/screens/my-courses/dashboard/add-subject-dropdown.tsx

```diff
@@ -2,17 +2,30 @@ import React from 'react'
 import { Dropdown } from 'react-bootstrap'
 import { map, groupBy, some } from 'lodash'
 import TutorDropdown from '../../../components/dropdown'
-
+import { useAllCourses } from '../../../store/courses'
+import { useAvailableOfferings } from '../../../store/offering'
 import { Offering, ID } from '../../../store/types'
+import Scroller from '../../../helpers/scroll-to'
 
 interface AddSubjectsDropdownProps {
-    allOfferings: Offering[]
     displayedOfferings: Offering[]
     setDisplayedOfferingIds: React.Dispatch<React.SetStateAction<ID[]>>
 }
 
-const AddSubjectsDropdown: React.FC<AddSubjectsDropdownProps> = ({ allOfferings, displayedOfferings, setDisplayedOfferingIds }) => {
+const scroller = new Scroller()
+
+const AddSubjectsDropdown: React.FC<AddSubjectsDropdownProps> = ({ displayedOfferings, setDisplayedOfferingIds }) => {
+    const courses = useAllCourses()
+    const allOfferings = useAvailableOfferings(courses)
     const offeringsBySubject = groupBy(allOfferings, o => o.subject)
+
+    const onSelectOffering = (offeringId: ID) => {
+        // adding the subject block at the top
+        setDisplayedOfferingIds(prevState => [offeringId, ...prevState])
+        // then scroll to the top to see the new offering block
+        scroller.scrollToTop()
+    }
```

Comment:
scroller has a `deferred` option that will wait 10ms then scroll.  We should use that here because react set state happens asynchronously and the offering probably hasn't been added by the time scroll starts.